src/test/groovy/org/osiam/test/integration/AbstractIT.groovy

```diff
@@ -123,7 +132,30 @@ abstract class AbstractIT extends Specification {
         osiamConnector.retrieveAccessToken(Scope.ALL)
     }
 
+    def fetchEmail(String userNameAndPassword) {
+        final Properties properties = System.getProperties()
+        final Session session = Session.getDefaultInstance(properties)
+        store = (POP3Store) session.getStore(ServerSetup.PROTOCOL_POP3)
+        store.connect('localhost', 11110, userNameAndPassword, userNameAndPassword)
+        inbox = store.getFolder('INBOX')
+        inbox.open(Folder.READ_WRITE)
+        final Message[] messages = inbox.getMessages()
+
+        // mark all emails as delete
+        for (int i = 1; i <= inbox.getMessageCount(); i++) {
+            inbox.getMessage(i).setFlag(Flags.Flag.DELETED, true)
+        }
+
+        messages
+    }
+
     def cleanup() {
+
+        if (inbox != null && store != null) {
```

Comment:
I need to check this, because only a few tests need the fetch emails... maybe there is a better way to cleanup after only the tests how are using emails?